### ISSUE TITLE
Some sites don't work until darkreader is toggled off and back on

### ISSUE THREAD

**Issue by counterbeing** [darkreader/darkreader#2065]
This issue is to replace [a closed issue](https://github.com/darkreader/darkreader/issues/1914) and provide more information.

These are sites that darkreader seems to work on, but if darkreader is already enabled when you navigate to the page, some or all elements are no longer in dark mode. These are sites where simply toggling the darkreader extension it suddenly works as expected.

Please add any website URLs along with screenshots, and any additional info you have that fits this description. I will be adding mine below as they arise.

**Comment by counterbeing** [comment]
Site:  https://www.apollographql.com/docs/apollo-server/
Before Toggle: 
![image](https://user-images.githubusercontent.com/239798/75702784-5e38da00-5c73-11ea-9cc5-7686c7ac28af.png)

After the toggle: 
![image](https://user-images.githubusercontent.com/239798/75702809-6bee5f80-5c73-11ea-8928-d303cb906998.png)

Platform: Mac OS 10.15.3
Browser: Version 79.0.3945.130 (Official Build) (64-bit)
Dark Reader Version: 4.8.9

**Comment by UndeadDevel** [comment]
Openjfx.io has this issue about 50% of the time on pretty much all sites there; screenshots from https://openjfx.io/javadoc/13/javafx.graphics/javafx/scene/layout/StackPane.html
As you can see, before the toggle only the scrollbar on the right and the search field are dark (they aren't when I turn the extension off). This doesn't just happen in Dynamic mode but pretty much all modes, though I don't recall if I've specifically tested the Static mode...lmk if you need that info.
![before_toggle](https://user-images.githubusercontent.com/30495439/75759335-46be2780-5d2d-11ea-8e79-6c42ce0b06cf.jpg)
![after_toggle](https://user-images.githubusercontent.com/30495439/75759342-49b91800-5d2d-11ea-86a4-a2d742786bc2.jpg)

**Comment by counterbeing** [comment]
Platform: Mac OS 10.15.3
Browser: Version 79.0.3945.130 (Official Build) (64-bit)
Dark Reader Version: 4.8.9
Site: https://fivethirtyeight.com/
Before Toggle:  
![image](https://user-images.githubusercontent.com/239798/75794640-512cf100-5d2e-11ea-9d28-0fa71ad14403.png)
After Toggle: 
![image](https://user-images.githubusercontent.com/239798/75794693-643fc100-5d2e-11ea-896b-e4d1c810f63f.png)

**Comment by counterbeing** [comment]
I did a little bit of debugging, and have a guess as to something that's causing the issue. I took the live source of before and after toggle and diff'd them. Here's how that looks:

```diff
diff --git a/before.html b/after.html
index 5abea42..1b80135 100644
--- a/before.html
+++ b/after.html
@@ -76,22 +76,13 @@ select:-webkit-autofill {
 <link rel="dns-prefetch" href="//platform.twitter.com">
 <link rel="dns-prefetch" href="//s.w.org">
 <link rel="stylesheet" id="all-css-0" href="https://fivethirtyeight.com/_static/??/wp-includes/css/dist/block-library/style.min.css,/wp-content/themes/espn-fivethirtyeight/assets/css/fivethirtyeight.min.css?m=1583252366" type="text/css" media="all">
+<style class="darkreader darkreader--sync" media="screen"></style>
 <style id="wp-block-library-inline-css">
 .has-text-align-justify{text-align:justify;}
 </style><style class="darkreader darkreader--sync" media="screen"></style>
 <script>
 var DisneyOneID = {"config":{"clientId":"DATG-FIVETHIRTYEIGHT.WEB-PROD","langPref":"en-US","debug":false,"responderPage":"https:\/\/fivethirtyeight.com\/oneid-responder","cssOverride":"https:\/\/fivethirtyeight.com\/wp-content\/themes\/espn-fivethirtyeight\/assets\/css\/oneid.min.css","campaignName":"FiveThirtyEight"}};
 </script>
-<style class="darkreader darkreader--override" media="screen">.jfk-bubble {
-    background-color: #fffff8;
-}
-.vimvixen-hint {
-    border-color: #d9b736 !important;
-    color: #f6eacb !important;
-}
-::placeholder {
-    opacity: 0.5 !important;
-}</style>
 <script src="https://cdn.registerdisney.go.com/v2/outer/DisneyID.js"></script>
 <script>
 var ESPNTaboola = {"publisher":"abcnews-fivethirtyeight","type":"home"};
@@ -182,7 +173,16 @@ If you need any further help, ask in the CSS Customization forum.
 .fb_dialog{background:rgba(82, 82, 82, .7);position:absolute;top:-10000px;z-index:10001}.fb_dialog_advanced{border-radius:8px;padding:10px}.fb_dialog_content{background:#fff;color:#373737}.fb_dialog_close_icon{background:url(https://static.xx.fbcdn.net/rsrc.php/v3/yq/r/IE9JII6Z1Ys.png) no-repeat scroll 0 0 transparent;cursor:pointer;display:block;height:15px;position:absolute;right:18px;top:17px;width:15px}.fb_dialog_mobile .fb_dialog_close_icon{left:5px;right:auto;top:5px}.fb_dialog_padding{background-color:transparent;position:absolute;width:1px;z-index:-1}.fb_dialog_close_icon:hover{background:url(https://static.xx.fbcdn.net/rsrc.php/v3/yq/r/IE9JII6Z1Ys.png) no-repeat scroll 0 -15px transparent}.fb_dialog_close_icon:active{background:url(https://static.xx.fbcdn.net/rsrc.php/v3/yq/r/IE9JII6Z1Ys.png) no-repeat scroll 0 -30px transparent}.fb_dialog_iframe{line-height:0}.fb_dialog_content .dialog_title{background:#6d84b4;border:1px solid #365899;color:#fff;font-size:14px;font-weight:bold;margin:0}.fb_dialog_content .dialog_title>span{background:url(https://static.xx.fbcdn.net/rsrc.php/v3/yd/r/Cou7n-nqK52.gif) no-repeat 5px 50%;float:left;padding:5px 0 7px 26px}body.fb_hidden{height:100%;left:0;margin:0;overflow:visible;position:absolute;top:-10000px;transform:none;width:100%}.fb_dialog.fb_dialog_mobile.loading{background:url(https://static.xx.fbcdn.net/rsrc.php/v3/ya/r/3rhSv5V8j3o.gif) white no-repeat 50% 50%;min-height:100%;min-width:100%;overflow:hidden;position:absolute;top:0;z-index:10001}.fb_dialog.fb_dialog_mobile.loading.centered{background:none;height:auto;min-height:initial;min-width:initial;width:auto}.fb_dialog.fb_dialog_mobile.loading.centered #fb_dialog_loader_spinner{width:100%}.fb_dialog.fb_dialog_mobile.loading.centered .fb_dialog_content{background:none}.loading.centered #fb_dialog_loader_close{clear:both;color:#fff;display:block;font-size:18px;padding-top:20px}#fb-root #fb_dialog_ipad_overlay{background:rgba(0, 0, 0, .4);bottom:0;left:0;min-height:100%;position:absolute;right:0;top:0;width:100%;z-index:10000}#fb-root #fb_dialog_ipad_overlay.hidden{display:none}.fb_dialog.fb_dialog_mobile.loading iframe{visibility:hidden}.fb_dialog_mobile .fb_dialog_iframe{position:sticky;top:0}.fb_dialog_content .dialog_header{background:linear-gradient(from(#738aba), to(#2c4987));border-bottom:1px solid;border-color:#043b87;box-shadow:white 0 1px 1px -1px inset;color:#fff;font:bold 14px Helvetica, sans-serif;text-overflow:ellipsis;text-shadow:rgba(0, 30, 84, .296875) 0 -1px 0;vertical-align:middle;white-space:nowrap}.fb_dialog_content .dialog_header table{height:43px;width:100%}.fb_dialog_content .dialog_header td.header_left{font-size:12px;padding-left:5px;vertical-align:middle;width:60px}.fb_dialog_content .dialog_header td.header_right{font-size:12px;padding-right:5px;vertical-align:middle;width:60px}.fb_dialog_content .touchable_button{background:linear-gradient(from(#4267B2), to(#2a4887));background-clip:padding-box;border:1px solid #29487d;border-radius:3px;display:inline-block;line-height:18px;margin-top:3px;max-width:85px;padding:4px 12px;position:relative}.fb_dialog_content .dialog_header .touchable_button input{background:none;border:none;color:#fff;font:bold 12px Helvetica, sans-serif;margin:2px -12px;padding:2px 6px 3px 6px;text-shadow:rgba(0, 30, 84, .296875) 0 -1px 0}.fb_dialog_content .dialog_header .header_center{color:#fff;font-size:16px;font-weight:bold;line-height:18px;text-align:center;vertical-align:middle}.fb_dialog_content .dialog_content{background:url(https://static.xx.fbcdn.net/rsrc.php/v3/y9/r/jKEcVPZFk-2.gif) no-repeat 50% 50%;border:1px solid #4a4a4a;border-bottom:0;border-top:0;height:150px}.fb_dialog_content .dialog_footer{background:#f5f6f7;border:1px solid #4a4a4a;border-top-color:#ccc;height:40px}#fb_dialog_loader_close{float:left}.fb_dialog.fb_dialog_mobile .fb_dialog_close_button{text-shadow:rgba(0, 30, 84, .296875) 0 -1px 0}.fb_dialog.fb_dialog_mobile .fb_dialog_close_icon{visibility:hidden}#fb_dialog_loader_spinner{animation:rotateSpinner 1.2s linear infinite;background-color:transparent;background-image:url(https://static.xx.fbcdn.net/rsrc.php/v3/yD/r/t-wz8gw1xG1.png);background-position:50% 50%;background-repeat:no-repeat;height:24px;width:24px}@keyframes rotateSpinner{0%{transform:rotate(0deg)}100%{transform:rotate(360deg)}}
 .fb_iframe_widget{display:inline-block;position:relative}.fb_iframe_widget span{display:inline-block;position:relative;text-align:justify}.fb_iframe_widget iframe{position:absolute}.fb_iframe_widget_fluid_desktop,.fb_iframe_widget_fluid_desktop span,.fb_iframe_widget_fluid_desktop iframe{max-width:100%}.fb_iframe_widget_fluid_desktop iframe{min-width:220px;position:relative}.fb_iframe_widget_lift{z-index:1}.fb_iframe_widget_fluid{display:inline}.fb_iframe_widget_fluid span{width:100%}</style>
 <style class="darkreader darkreader--sync" media="screen"></style>
-</head>
+<style class="darkreader darkreader--override" media="screen">.jfk-bubble {
+    background-color: #fffff8;
+}
+.vimvixen-hint {
+    border-color: #d9b736 !important;
+    color: #f6eacb !important;
+}
+::placeholder {
+    opacity: 0.5 !important;
+}</style></head>
 <body class="home page-template-default page page-id-10521 slug-home touch">
 
 <a href="#content" class="skip-to-content-link visually-hidden">Skip to main content</a>
```

I realize the full context would be more helpful, but I didn't want to overwhelm the thread with html. So i'll  just point out the possibly important bit: there are html elements which seem to move around within the head tag. So maybe this is an ordering issue?

**Comment by alexanderby** [comment]
Thanks for the detailed report. I will try to understand, why some stylesheets are not processed.

**Comment by prg318** [comment]
Additional sites with this issue:
archlinux.org
before:
![archlinux-before](https://user-images.githubusercontent.com/549291/75911352-70567c00-5e1d-11ea-8259-6c22b53cfdc4.png)
after:
![archlinux-after](https://user-images.githubusercontent.com/549291/75911397-7ea49800-5e1d-11ea-9a4e-38f5f21afaa8.png)

outlook web access
before:
![owa-before](https://user-images.githubusercontent.com/549291/75911830-2e7a0580-5e1e-11ea-899d-2cf9a0738fdb.png)
after:
![owa-after](https://user-images.githubusercontent.com/549291/75911840-320d8c80-5e1e-11ea-9ae3-9a2cf2dee96a.png)

Both tested with:
Firefox 73.0.1
Dark Reader 4.8.9
Arch Linux

**Comment by Jakosaur** [comment]
Hopefully related, sorry if not. Some aspects changed to dark mode, but others aren't until toggled.

Edge Chromium, Stable
Dark Reader 4.8.9
Windows 10

[https://beta.support.xbox.com](https://beta.support.xbox.com)
[https://beta.support.xbox.com/help/hardware-network/warranty-service/getting-your-product-serviced](https://beta.support.xbox.com/help/hardware-network/warranty-service/getting-your-product-serviced)

Before:
![image](https://user-images.githubusercontent.com/20144356/75984919-33ed5380-5ee3-11ea-83ca-5a42acc853c8.png)

After:
![image](https://user-images.githubusercontent.com/20144356/75984939-4071ac00-5ee3-11ea-9985-5931d59a52bd.png)

Before:
![image](https://user-images.githubusercontent.com/20144356/75984955-49fb1400-5ee3-11ea-8fd4-879ccff2c168.png)

After:
![image](https://user-images.githubusercontent.com/20144356/75984968-4cf60480-5ee3-11ea-9c13-c5f1fee282db.png)

**Comment by dark-nova** [comment]
I have seen partial breakage on https://docs.conda.io, but only with the Chrome version. I also tested on Firefox ESR, but Firefox seems unaffected. Both Chrome and Firefox extensions are version 4.8.10.

https://docs.conda.io/en/latest/miniconda.html
<details><summary>Before</summary>
<img src="https://user-images.githubusercontent.com/31264514/76556100-25a8b580-6456-11ea-8e0d-c1995f6b0697.png" />
</details>
<details><summary>After</summary>
<img src="https://user-images.githubusercontent.com/31264514/76556156-3fe29380-6456-11ea-8a74-f328c0d0c4e5.png" />
</details>

https://docs.conda.io/en/latest/index.html
<details><summary>Before</summary>
<img src="https://user-images.githubusercontent.com/31264514/76556296-7fa97b00-6456-11ea-8f6c-509e5b55261b.png" />
</details>
<details><summary>After</summary>
<img src="https://user-images.githubusercontent.com/31264514/76556326-89cb7980-6456-11ea-8c03-3148c83a9cb8.png" />
</details>

It almost feels like the CSS loads too slowly - on Firefox, the page loads slower than on Chrome.

**Comment by amanhigh** [comment]
On investing.com issue has reappeared.

First Time Opening Site
![SNAG--2020-03-13-022](https://user-images.githubusercontent.com/409973/76603751-8ce86780-6533-11ea-8583-dbdc1507b15f.jpg)

Disabling Site
![SNAG--2020-03-13-025](https://user-images.githubusercontent.com/409973/76603772-95d93900-6533-11ea-9837-bca53c098b1d.jpg)

Re-enabling Site
![SNAG--2020-03-13-024](https://user-images.githubusercontent.com/409973/76603756-8eb22b00-6533-11ea-8f49-4489f1646de3.jpg)

**Comment by ElhemEnohpi** [comment]
On www.pewforum.org pages, most of the text is black on a dark background, the page footer isn't inverted, etc., until Dark Reader is toggled off and on again. Then everything is fine (except the masthead logo isn't inverted, but that can be fixed in developer tools).

Tested with:
Dark Reader 4.8.10 on Waterfox - Mozilla/5.0 (Macintosh; Intel Mac OS X 10.14; rv:56.0) Gecko/20100101 Firefox/56.0 Waterfox/56.3
Dark Reader 4.9.2 on Firefox - Mozilla/5.0 (Macintosh; Intel Mac OS X 10.14; rv:74.0) Gecko/20100101 Firefox/74.0
Dark Reader 4.9.2 on Vivaldi - Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.158 Safari/537.36 Vivaldi/2.11.1811.49

**Comment by TheRealPSV** [comment]
I'm not sure when this started happening, but I know it wasn't always an issue. I'm getting this all over the place. One example I've noticed lately is https://forums.dolphin-emu.org/

Is anyone willing to try older versions to see which version broke it?

**Comment by Racle** [comment]
Few related issues:
https://github.com/darkreader/darkreader/issues/1224
https://github.com/darkreader/darkreader/issues/1806

**Comment by Gusted** [comment]
the archlinux one has been kind of fixed, the darkreader-fallback wasn't setted the first time but after manually adding into the dev tools the site worked again https://github.com/darkreader/darkreader/issues/1224#issuecomment-614513974 for more info. Will look into if the other sites are the same issue.

**Comment by Gusted** [comment]
https://apollographql.com/docs/apollo-server -> Doesn't work Only after Toggling. Firefox Display the good content but after some seconds the incorrect one
https://openjfx.io/javadoc/13/javafx.graphics/javafx/scene/layout/StackPane.html -> Can't reproduce with Firefox and Chromium Seems fixed?
https://fivethirtyeight.com/ -> Can't reproduced with Firefox and Chromium. Seems fixed?
https://archlinux.org/ -> Fixed with manual adding the fallback style. Not related to this kind of issue from the problem probaly as it is fully white instead of incorrect.
https://beta.support.xbox.com/ -> Related to this issue and still persist. Same incorrect results with Firefox and Chromium
https://docs.conda.io/en/latest/ -> Render issue? Firefox have no issues with this site. (Not related to main Issue Probaly)
https://investing.com/ -> Hmm Firefox only Issue. Chromium no problems with. (Not related to main Issue Probaly)
https://www.pewforum.org/ -> Related to this issue and still persist. Same incorrect results with Firefox and Chromium.
https://bbc.co.ok/news -> Can't reproduced with Firefox and Chromium. Seems fixed?
https://forum.tz-uk.com/forum.php -> Can't reproduced with Firefox and Chromium. Seems fixed?
https://forums.watchuseek.com/forum.php -> Can't reproduced with Firefox and Chromium. Seems fixed? Can use some Manual CSS to fix some color issues but not related to this issue.
https://forums.dolphin-emu.org/ -> Can't reproduced with Firefox and Chromium. Seems fixed?

For the Related Issue:
https://github.com/darkreader/darkreader/issues/1224#issuecomment-614513974 See that for #1224
#1806 is just firefox related and not both browsers like in this issue But could be the same origion of problem.

Tested with:
Firefox Nightly 77.0a1 (2020-04-15) (64-bit)
Brave Beta Version 1.8.75 Chromium: 81.0.4044.92 (Official Build) beta (64-bit
Darkreader local sideloaded as of https://github.com/darkreader/darkreader/tree/5b296478e831e71379fb065381faa79a298b9e39

**Comment by ForgeSource** [comment]
Thanks, but sadly it's still broken for me on the 3 sites that I reported (and on many others).

I'm using Waterfox Classic 2020.04 and DarkReader 4.9.4.

**Comment by Gusted** [comment]
Could be that's a waterfox Related Issue gonna test soon

**Comment by aliakbarmostafaei** [comment]
Can confirm this on Medium articles containing code block sections:

![Screen Shot 1399-02-14 at 01 18 57](https://user-images.githubusercontent.com/18480836/80891874-233b3d00-8cdc-11ea-8fa7-c2cf3db6dda0.png)

Using latest Safari 13.1 and Dark Reader 1.3.3.

**Comment by Gusted** [comment]
> Can confirm this on Medium articles containing code block sections, these are LAZY-Loaded only after scrolling down to their location:
> 
> ![Screen Shot 1399-02-14 at 01 18 57](https://user-images.githubusercontent.com/18480836/80891874-233b3d00-8cdc-11ea-8fa7-c2cf3db6dda0.png)
> 
> Using latest Safari 13.1 and Dark Reader 1.3.3.

Safari is closed source

**Comment by hotaru2k3** [comment]
seeing this in both Chrome and Firefox (screenshots are from Chrome) on cloud.linode.com:

before toggle:
<img width="1920" alt="2020-05-10" src="https://user-images.githubusercontent.com/312461/81509036-e1ab2300-92d5-11ea-970e-1c0306659115.png">

after toggle:
<img width="1920" alt="2020-05-10 (1)" src="https://user-images.githubusercontent.com/312461/81509037-e8d23100-92d5-11ea-8082-672ddc50f213.png">

**Comment by bherila** [comment]
This one too: 
* https://everythingplace.cargo.site/

**Comment by lonix1** [comment]
also
- [news.google.com](https://news.google.com)
- almost every other site I use

The only firefox shortcut I remember is the one to toggle DarkReader :)

**Comment by amanhigh** [comment]
> The only firefox shortcut I remember is the one to toggle DarkReader :)
> 

I can so relate to it. Darkreader is great extension with colors better than any other extension but has become very unusable due to this issue. I have to constantly do alt+d on windows.

**Comment by Gusted** [comment]
> I can so relate to it. Darkreader is great extension with colors better than any other extension but has become very unusable due to this issue. I have to constantly do alt+d on windows.

As long you all report sites with this kind of issues I can resolve it. I'm getting to understand why this happening I just need to find the faulty code.
Simply: darkreader--sync isn't placed at first load after reloading it will be placed.

**Comment by Gusted** [comment]
Quick Note: Found the problem and fixed the issue, problem: It was pretty obvious how to fix I don't know if I have messed up any other sites or code..... I will explain in my next comment what happend and the faulty code, but this costed me a lot of debugging for little faulty code 😅

**Comment by counterbeing** [comment]
That's exciting @Gusted! Let me know if I can test it out for you at some point, or when it gets released, and I'll confirm and close the issue! Thanks so much!

**Comment by Gusted** [comment]
Hi their!

So bugs we have them a lot of darkreader but also fixed a lot, but as we don't always know if it mess up any site we just await reactions in the issues tab for broken code. 

So let's introduce the faulty code https://github.com/darkreader/darkreader/blob/master/src/inject/utils/dom.ts#L100 this here checks if the elements stil exist and if the parent's are different. Well now it's just give an warning it's been removed, but for a site like Apollo(Only tested to work on this as this is the OP's reported website) that FIRST load their styles into <html> <head> a lot of styles here </head> </html> and after DOMContentload(I think) it's moves them to another location and that's cause to be the parent's not the same anymore but the <style> still exist and the --sync doesn't move, but after reloading darkreader and doesn't had to check if it's moved(as it was on the good location already the <style>) it showed the better colors etc. 

And after replacing the code with
```
        if (prevSibling && prevSibling.parentNode !== parent) {
            // Moving to another location with not same parents anymore, but still exist.
            // https://github.com/darkreader/darkreader/issues/2065
            logWarn('Style was moved to another location, restoring node\'s position', node, prevSibling, parent);
            prevSibling.parentNode.insertBefore(node, prevSibling.nextSibling);
            onRestore && onRestore();
            return;
        }
``` 
It will just move to the new location of that style(Tricky part as I think actually removed styles can mess up with this...

I didn't tested it yet for the other sites as they also can have different issues like here https://github.com/darkreader/darkreader/issues/2065#issuecomment-614572934 explained..... 

[darkreader.zip](https://github.com/darkreader/darkreader/files/4640373/darkreader.zip)
Chrome & Firefox builded code ^ Goodluck sideloading and report any bugs etc here. Before merging this code

Regards,
Gusted

**Comment by Gusted** [comment]
> That's exciting @Gusted! Let me know if I can test it out for you at some point, or when it gets released, and I'll confirm and close the issue! Thanks so much!

https://github.com/darkreader/darkreader/files/4640373/darkreader.zip
Perhabs darkreader v5 or maybe 4.9.9 if their is an critical issue

**Comment by counterbeing** [comment]
@Gusted I can confirm, this appears to fix the issue on a couple of sites I noticed the problem. Nice work!

**Comment by Gusted** [comment]
> @Gusted I can confirm, this appears to fix the issue on a couple of sites I noticed the problem. Nice work!

Nice to hear! Which sites still doesn't work what you have tested

**Comment by counterbeing** [comment]
The apollo site works great now.

If you go to https://fivethirtyeight.com/ and you click a link, you'll notice you then have to toggle the page to get it to work again.

**Comment by counterbeing** [comment]
I also see an error: 
![image](https://user-images.githubusercontent.com/239798/82160732-5e557880-9865-11ea-8184-9e62f67c3453.png)

**Comment by Gusted** [comment]
> I also see an error:
> ![image](https://user-images.githubusercontent.com/239798/82160732-5e557880-9865-11ea-8184-9e62f67c3453.png)

Ah didn't tested it on errors yet...

**Comment by counterbeing** [comment]
archlinux.org and investing.com both appear to work now. I know there are others that i run into regularly, i'll run this for a while and see what i see.

**Comment by Gusted** [comment]
> archlinux.org and investing.com both appear to work now. I know there are others that i run into regularly, i'll run this for a while and see what i see.

archlinux had an CSP problem that already had been fixed so far I knew...
Feel free to report bugs with it etc. I will have a quick look at the error. As I already expected to have some error with the code in some way.

**Comment by Gusted** [comment]
> The apollo site works great now.
> 
> If you go to https://fivethirtyeight.com/ and you click a link, you'll notice you then have to toggle the page to get it to work again.

Declare `a link`. Seems like it's working for me


And here is the version with the error checking for tutorial/resolvers/ even didn't got to reproduce it....
[darkreader.zip](https://github.com/darkreader/darkreader/files/4641152/darkreader.zip)

**Comment by counterbeing** [comment]
https://fivethirtyeight.com/features/how-much-money-could-student-athletes-make-as-social-media-influencers/ <https://fivethirtyeight.com/features/how-much-money-could-student-athletes-make-as-social-media-influencers/>

For example, when it first loads the page, it’s a light background (chrome Mac).

**Comment by Gusted** [comment]
Ah I see the light background but that's due to performance issues(Observing until injecting style can take long that is it noticable).... not related to main issue.

**Comment by counterbeing** [comment]
Just to be clear, it stays white, it never goes dark.

**Comment by Gusted** [comment]
Will look into that this/next week.
Are their any errors that showed up?

**Comment by counterbeing** [comment]
Actually, yes, I saw the same error.

**Comment by Gusted** [comment]
Test it with the updated extension I posted that should fixed it else I need to get it reproduced somehow. 

> [darkreader.zip](https://github.com/darkreader/darkreader/files/4641152/darkreader.zip)

**Comment by Racle** [comment]
Just for update, this doesn't fix site what I'm developing.
It's annoyingly one of the few sites which doesn't work and which I cannot show :/ 
I'll try if I can create demosite for this so might be easier to debug.

What I can see now, is that it doesn't apply any background-color / color to any elements until I've toggled extension.
`<style>` -tags are added to `<head>` -tag on initial load. Element inline styles doesn't work until toggle.


My original issue: https://github.com/darkreader/darkreader/issues/1806#issuecomment-614585711

**Comment by Gusted** [comment]
> Just for update, this doesn't fix site what I'm developing.
> It's annoyingly one of the few sites which doesn't work and which I cannot show :/
> I'll try if I can create demosite for this so might be easier to debug (might take some time as I have to splice up huge project or start from scratch)
> 
> What I can see now, is that it doesn't apply any background-color / color to any elements until I've toggled extension.
> `<style>` -tags are added to `<head>` -tag on initial load. Element inline styles doesn't work until toggle.
> 
> My original issue: [#1806 (comment)](https://github.com/darkreader/darkreader/issues/1806#issuecomment-614585711)

Did you got any errors in the console of that site or in the extension background.html itself? If no than it's most likely another issue and not related to moved to another parent element. @Racle 
A demosite would be appricate to better understand the issue.

**Comment by Racle** [comment]
@Gusted 
From background/index.html I get this
```sh
Security Error: Content at moz-extension://97d337bb-ee8f-4934-920b-dc265039dcc2/background/index.html may not load data from blob:https://oursite/3510cae6-7403-4a90-816b-e07849f2a6d9.
```
There is multiple blobs (20+) which throws that error.

After error lines I get:
```
Creating CSS for url: https://dev.hausvise.fi/hausvise/dashboard
``` 

that blob contains css and JS for our site which is loaded when I go to the site.


Toggling addon doesn't cause Security errors anymore, so it's only happening while loading site.

**Comment by Racle** [comment]
@Gusted 
From background/index.html I get this
```sh
Security Error: Content at moz-extension://97d337bb-ee8f-4934-920b-dc265039dcc2/background/index.html may not load data from blob:https://oursite/3510cae6-7403-4a90-816b-e07849f2a6d9.
```
There is multiple blobs (20+) which throws that error.

After error lines I get:
```
Creating CSS for url: https://oursite/hausvise/dashboard
``` 

that blob contains css and JS for our site which is loaded when I go to the site.


Toggling addon doesn't cause Security errors anymore, so it's only happening while loading site.

**Comment by Racle** [comment]
@Gusted
From background/index.html I get this

Security Error: Content at moz-extension://97d337bb-ee8f-4934-920b-dc265039dcc2/background/index.html may not load data from blob:https://oursite/3510cae6-7403-4a90-816b-e07849f2a6d9.
There is multiple blobs (20+) which throws that error.

After error lines I get:

Creating CSS for url: https://oursite/dashboard
that blob contains css and JS for our site which is loaded when I go to the site.

Toggling addon doesn't cause Security errors anymore, so it's only happening while loading site.

**Comment by Gusted** [comment]
@Racle CSP? Open a seperate issue, CSP is a nightmare case with extension etc.

**Comment by Racle** [comment]
> @Racle CSP? Open a seperate issue, CSP is a nightmare case with extension etc.

setting `security.csp.enable` `false` manually in Firefox doesn't affect in any way. Same error message still showing up. On chrome everything works perfectly.

**Comment by Gusted** [comment]
> setting `security.csp.enable` `false` manually in Firefox doesn't affect in any way. Same error message still showing up. On chrome everything works perfectly.

Firefox & CSP isn't the best implemantation is too over protected so probably setting CSP off it still cause erorrs. Chrome works as it's a better implementation in my opinion.

**Comment by Racle** [comment]
Running code straight from without nginx doesn't change situation.
Webpack and expressJs doesn't set CSP automatically, and we have not set that in that state.

Thereforce I doubt that CSP is issue here.

**Comment by Gusted** [comment]
> Running code straight from without nginx doesn't change situation.
> Webpack and expressJs doesn't set CSP automatically, and we have not set that in that state.
> 
> Thereforce I doubt that CSP is issue here.

Is the content loaded from external site that has csp?
BTW at which line was the error

**Comment by Racle** [comment]
> Is the content loaded from external site that has csp?
> BTW at which line was the error

All is loaded from localhost (when running locallu) or on our developement server (when running remotely without debug tools)

Security error doesn't show line number.

**Comment by Gusted** [comment]
> All is loaded from localhost (when running locallu) or on our developement server (when running remotely without debug tools)
> 
> Security error doesn't show line number.

Okay, then we need to modify our code again smh, I will need a POC it don't needs to be a huge POC just the code to get the CSP triggered and see If I can determine what darkreader is doing or actually need to report to Mozilla as blob: don't had any CSP restrictions so far I knew with chrome

**Comment by counterbeing** [comment]
Thanks so much for everyone's work on this! Just made a small contribution in gratitude of saving my eyeballs!  🎉

**Comment by Gusted** [comment]
We are glad to here that :)

**Comment by Racle** [comment]
> Okay, then we need to modify our code again smh, I will need a POC it don't needs to be a huge POC just the code to get the CSP triggered and see If I can determine what darkreader is doing or actually need to report to Mozilla as blob: don't had any CSP restrictions so far I knew with chrome

@Gusted Made quick POC on my web-server, but rather not send that out publicly (as I didn't have time to properly clean that code :) )

Any way to send you a private message where I can share that url so you can check it out?

**Comment by Gusted** [comment]
`williamzijl7@hotmail.com` Email is already public trough patch files om my commits 🙄
